internal/metrics/collectors/controller.go

```diff
@@ -7,43 +7,79 @@ var labelNamesController = []string{"type"}
 // ControllerCollector is an interface for the metrics of the Controller
 type ControllerCollector interface {
 	SetIngressResources(ingressType string, count int)
+	SetVsResources(count int)
```

Comment:
For consistency with the existing method:
SetVsResources -> SetVirtualServerResources
SetVsrResources -> SetVirtualServerRouteResources

You can also remove resources in all three methods and make them plural to be less verbose:
SetVirtualServers

Reply:
Please also update `updateVsMetrics` in Controller` to `updateVirtualServerMetrics` for the same reasons.